src/lib/MapPreviewModal.svelte

```diff
@@ -0,0 +1,56 @@
+<script lang="ts">
+  export let key: string | null = null
+
+  const close = () => (key = null)
+  const onKeydown = (e: KeyboardEvent) => {
+    if (e.key === 'Escape') {
+      close()
+      document.removeEventListener('keydown', onKeydown)
+    }
+  }
+  document.addEventListener('keydown', onKeydown)
+</script>
+
+<div class="preview-container">
+  <div class="backdrop" on:click={close} />
+  <div class="content">
+    <iframe
+      src="https://allpoland.github.io/ArcViewer/?id={key}"
+      title="Map Preview"
+      height="100%"
+      width="100%"
+      allowfullscreen="true"
+    />
+  </div>
+</div>
+
+<style lang="scss">
+  .preview-container {
+    display: flex;
+    z-index: 100;
+    justify-content: center;
+    align-items: center;
+    position: fixed;
+    top: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+
+    .backdrop {
+      position: fixed;
+      top: 0;
+      left: 0;
+      height: 100vh;
```

Comment:
This should be less prone to cause accidental overflows:
```suggestion
      bottom: 0;
```